From ae647879efad6ba2bb0b95ed7e70dee82c05b82a Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Fri, 2 Jun 2017 07:40:08 +0300
Subject: [PATCH] suricata_hawkbit: depends on HAVE_JSON_C

Unconditionally selecting JSON when !HAVE_JSON_C leads to a warning:

warning: (SURICATTA_HAWKBIT) selects JSON which has unmet direct dependencies (HAVE_JSON_C)

and eventually a build failure:

In file included from corelib/artifacts_versions.c:34:0:
include/parselib.h:61:25: fatal error: json-c/json.h: No such file or directory

Make SURICATTA_HAWKBIT depend on HAVE_JSON_C.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status:
https://groups.google.com/forum/?hl=en#!topic/swupdate/WmwNN_Hni0c

 suricatta/Config.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/suricatta/Config.in b/suricatta/Config.in
index 24bd5d1053f5..cfd96b83506a 100644
--- a/suricatta/Config.in
+++ b/suricatta/Config.in
@@ -54,6 +54,7 @@ choice
 config SURICATTA_HAWKBIT
 	bool "hawkBit support"
 	depends on HAVE_LIBCURL
+	depends on HAVE_JSON_C
 	select JSON
 	select SURICATTA
 	help
-- 
2.11.0

